src/v/raft/vote_stm.cc

```diff
@@ -265,6 +265,8 @@ ss::future<> vote_stm::update_vote_state(ss::semaphore_units<> u) {
                 "{} - {} ",
                 ec.value(),
                 ec.message());
+          } else {
```

Comment:
In this branch, it is expected that _confirmed_term will be == _term once replicating config is complete, right?  Maybe an assertion would be good.

Alternatively, we could just set `_confirmed_term` here instead of in do_maybe_update_leader_commit_idx

Reply:
Good point, added